specutils/manipulation/smoothing.py

```diff
@@ -43,11 +53,51 @@ def convolution_smooth(spectrum, kernel):
     # Smooth based on the input kernel
     smoothed_flux = convolution.convolve(flux, kernel)
 
+    # Propagate the uncertainty if it exists...
+    uncertainty = copy.deepcopy(spectrum.uncertainty)
+    if uncertainty is not None:
+        if isinstance(uncertainty, StdDevUncertainty):
+            # Convert
+            values = uncertainty.array
+            ivar_values = 1 / values**2
+
+            # Propagate
+            prop_ivar_values = convolution.convolve(ivar_values, kernel)
+
+            # Put back in
+            uncertainty.array = 1 / np.sqrt(prop_ivar_values)
+
+        elif isinstance(uncertainty, VarianceUncertainty):
+            # Convert
+            values = uncertainty.array
+            ivar_values = 1 / values
+
+            # Propagate
+            prop_ivar_values = convolution.convolve(ivar_values, kernel)
+
+            # Put back in
+            uncertainty.array = 1 / prop_ivar_values
+
+        elif isinstance(uncertainty, InverseVariance):
+            # Convert
+            ivar_values = uncertainty.array
+
+            # Propagate
+            prop_ivar_values = convolution.convolve(ivar_values, kernel)
+
+            # Put back in
+            uncertainty.array = prop_ivar_values
+        else:
+            warnings.warn("Uncertainty is {} but convolutional error propagation is not defined for that type.".format(type(uncertainty)),
```

Comment:
This should also actually say what happens?  I.e.:
```suggestion
            warnings.warn("Uncertainty is {} but convolutional error propagation is not defined for that type. Uncertainty will be passed through un-altered.".format(type(uncertainty)),
```

But arguably, might we actually want to *drop* the uncertainty since we don't know what to do with it?  I.e.,
```suggestion
            uncertainty = None
            warnings.warn("Uncertainty is {} but convolutional error propagation is not defined for that type. Uncertainty will be dropped in the convolved spectrum.".format(type(uncertainty)),
```

Reply:
I am fine with setting the uncertainty to None.